README.md

```diff
@@ -134,12 +134,24 @@ The file has three main sections.
 
 The `services` section is a hash, where the keys are the `name` of the service to be configured.
 The name is just a human-readable string; it will be used in logs and notifications.
-Each value in the services hash is also a hash, and should contain the following keys:
+Each value in the services hash is also a hash, and must contain the following keys:
+
+* [`discovery`](#discovery): how synapse will discover hosts providing this service (see [below](#discovery))
+
+The services hash *should* contain a section on how to configure the routing
+component you wish to use for this particular service. The only choice currently
+is `haproxy`:
 
-* [`discovery`](#discovery): how synapse will discover hosts providing this service (see below)
-* `default_servers`: the list of default servers providing this service; synapse uses these if no others can be discovered
 * [`haproxy`](#haproxysvc): how will the haproxy section for this service be configured
 
+The services hash may contain the following keys:
+
+* `default_servers` (default: `[]`): the list of default servers providing this service; synapse uses these if no others can be discovered. See [Listing Default Servers](#defaultservers).
+* `keep_default_servers` (default: false): whether default servers should be added to discovered services
+* `use_previous_backends` (default: true): if at any time the registry drops all backends, use previous backends we already know about.
+<a name="backend_port_override"/>
+* `backend_port_override`: the port that discovered servers listen on; you should specify this if your discovery mechanism only discovers names or addresses (like the DNS watcher or the Ec2TagWatcher). If the discovery method discovers a port along with hostnames (like the zookeeper watcher) this option may be left out, but will be used in preference if given.
+
```

Comment:
I've noticed that this isn't actually a full list. Let's document all the options that we can set (filters, leader_election, etc ...)